src/server/src/modules/student.ts

```diff
@@ -128,3 +127,52 @@ export const getJobApplications = async (studentId: number) => {
         ])
         .getMany();
 };
+
+export const searchStudents = async (
+    firstName: string,
+    lastName: string,
+    email: string,
+    sid: string,
+    departmentIds: number[],
+    classStandings: classStandings[],
+    page: number,
+    numOfItems: number
+) => {
+    return await getRepository(Student)
+        .createQueryBuilder('student')
+        .leftJoin('student.user', 'user')
+        .addSelect([
+            'user.id',
+            'user.firstName',
+            'user.lastName',
+            'user.middleName',
+            'user.biography',
+            'user.email',
+        ])
+        .leftJoinAndSelect('student.department', 'department')
+        .leftJoinAndSelect('department.college', 'college')
+        .leftJoinAndSelect('student.courses', 'courses')
+        .where('LOWER(user.firstName) LIKE :firstName', {
+            firstName: `%${firstName.toLowerCase()}%`,
+        })
+        .andWhere('LOWER(user.lastName) LIKE :lastName', {
+            lastName: `%${lastName.toLowerCase()}%`,
+        })
+        .andWhere('user.email LIKE :email', {
+            email: `%${email}%`,
+        })
+        .andWhere('((NOT :sidExists) OR (:sidExists AND sid LIKE :sid))', {
+            sidExists: sid == "" ? false : true,
+            sid: `%${sid}%`,
+        })
+        .andWhere('(NOT :departmentIdsPopulated OR department.id IN (:...departmentIds))', {
+            departmentIdsPopulated: departmentIds[0] != -1,
```

Comment:
if departmentIds come as an empty array, I think departmentIds[0] will return undefined which is not -1, so you will get a true for your departmentIdsPopulated variable.  Maybe I better check would be departmentIds.length > 0. Also in typescript equality is === and inequality is !==

Reply:
When departmentIds is empty, it causes an sql parse error when I do "department.id IN (:...departmentIds)"
Therefore, when departmentIds is empty, I populate it with -1. So currently, departmentIds is never empty.
Please let me know if you think of another solution.

Reply:
Done.